src/util/io.c

```diff
@@ -140,33 +143,55 @@ socket_close (
     return TSS2_RC_SUCCESS;
 }
 
+static void uinttostring(unsigned int num, char *str) {
+    int len = 0;
+    unsigned int n;
+ 
+    for (n = num; n != 0; ++len, n /= 10) ;
+
+    for (int i = 1, n = num; i <= len; n /= 10, ++i) {
+        str[len - i] = '0' + (n % 10);
+    }
+    str[len] = '\0';
+}
+
 TSS2_RC
 socket_connect (
     const char *hostname,
     uint16_t port,
     SOCKET *sock)
 {
-    struct sockaddr_in sockaddr = { 0 };
+    static const struct addrinfo hints = { .ai_socktype = SOCK_STREAM,
+        .ai_protocol = IPPROTO_TCP };
```

Comment:
Just for clearity, please set .ai_family = AF_UNSPEC

Reply:
OK. That's the default (0), as with all fields, but it does make it explicit.